content/en/blog/_posts/2023-12-13-Canary-Deployment-With-Kubernetes/index.md

```diff
@@ -0,0 +1,336 @@
+docker push <your Docker Hub username>/canary-app:v1
+```
+
+Now modify text inside the `index.html` file using this command:
+
+```bash
+echo "Hello from app-v2" > index.html
+```
+Create Docker image for canary version of app, using a `v2` tag by running
+this command:
+
+```bash
+docker build . -t <your Docker Hub username>/canary-app:v2
+```
+
+Push the canary version container image to the Docker Hub using this command:
+
+```bash
+docker push <your Docker Hub username>/canary-app:v2
+```
+
+## Deploy the app
+
+To deploy the app, create a `deployment.yaml` manifest file with the following
+configuration, and replace your Docker Hub username in the `image` field.
+
+```yaml
+# deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: canary-app-deployment
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: canary-app
+      version: v1
+  template:
+    metadata:
+      labels:
+        app: canary-app
+        version: v1
+    spec:
+      containers:
+        - name: canary-app
+          image: <your Docker Hub username>/canary-app:v1
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: canary-app-deployment-v2
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: canary-app
+      version: v2
+  template:
+    metadata:
+      labels:
+        app: canary-app
+        version: v2
+    spec:
+      containers:
+        - name: canary-app
+          image: <your Docker Hub username>/canary-app:v2
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: canary-app-service
+spec:
+  type: LoadBalancer
+  selector:
+    app: canary-app
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+```
+
+This Kubernetes manifest deploys two different apps, `canary-app:v1`, `canary-app:v2`
+and a single service for both of the apps. The service is exposed via a `LoadBalancer`.
+
+To access the application running inside the pod you need to get the IP of the
+load balancer. To get the IP of load balancer simply run `kubectl get services`
+you will see the `canary-app-service` of type `LoadBalancer`. This IP will be used
+in the next command. And if you are using Minikube and see the IP of load balancer
+is in pending state, just run the `minikube tunnel` command in the background.
+And then run `kubectl get services`.
+
+{{< figure src="load-balancer-ip.png" alt="A screenshot of a terminal window showing external IP of load balancer service." >}}
+
+Now run this command to make requests to the load-balancer
+(replace the ip of the load-balancer in the command):
+
+```bash
+for i in {1..10}; do curl <ip of load-balancer service>; done
+```
+
+After running this command you will get a similar output like the following.
+
+{{< figure src="curl-to-load-balancer.png" alt="A screenshot of terminal window showing output of making CURL request to the load balancer. The output contains `Hello from app-v1` 4 times and `Hello from app-v2` 6 times">}}
```

Comment:
```suggestion
{{< figure src="curl-to-load-balancer.png" alt="A screenshot of terminal window showing output of making CURL request to the load balancer. The output contains 'Hello from app-v1' 4 times and 'Hello from app-v2' 6 times" >}}
```

Can we use text here rather than a screenshot?